src/envs/painting.py

```diff
@@ -54,10 +60,10 @@ def __init__(self) -> None:
             "pose_x", "pose_y", "pose_z", "dirtiness", "wetness", "color",
             "grasp", "held"
         ])
-        self._box_type = Type("box", ["color"])
+        self._box_type = Type("box", ["pose_x", "pose_y", "color"])
         self._lid_type = Type("lid", ["is_open"])
-        self._shelf_type = Type("shelf", ["color"])
-        self._robot_type = Type("robot", ["fingers"])
+        self._shelf_type = Type("shelf", ["pose_x", "pose_y", "color"])
+        self._robot_type = Type("robot", ["pose_x", "pose_y", "fingers"])
```

Comment:
Looks like the robot's pose_y field here is not constant, which means its presence is going to affect the predicate invention pipeline. So definitely we should run the nightly before merging this.